### PR TITLE
Add BOM to the module

### PR DESCRIPTION
**Pull request by timyates**
Use catalog versions
Should be released as 3.2.0 for Micronaut 3.9.0

### CODE REVIEW

**Comment by timyates** [comment]
Raised a PR in core to add the bom for 3.9.0 once this is released https://github.com/micronaut-projects/micronaut-core/pull/8659

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -1,18 +1,7 @@
...
-micronaut-management = { module = "io.micronaut:micronaut-management" }
-micronaut-validation = { module = "io.micronaut:micronaut-validation" }
-
-reactor-core = { module = "io.projectreactor:reactor-core" }
-spock = { module = "org.testcontainers:spock" }
+micronaut-docs = "2.0.0"
+micronaut = "3.8.2"
+micronaut-test = "3.0.5"
+groovy = "3.0.8"
+spock = "2.2-groovy-3.0"
```
Comment (sdelamo) [inline gradle/libs.versions.toml:7]:
```suggestion
spock = "2.3-groovy-3.0"
```

File: discovery-client/build.gradle [hunk 2]
```diff
@@ -1,24 +1,24 @@
...
-    implementation(libs.reactor.core)
+    implementation(mn.reactor.core)
 
-    compileOnly(libs.micronaut.management)
+    compileOnly(mn.micronaut.management)
 
-    testAnnotationProcessor(libs.micronaut.inject.java)
-    testImplementation(libs.spock)
+    testAnnotationProcessor(mn.micronaut.inject.java)
+    testImplementation("org.testcontainers:spock")
```
Comment (sdelamo) [inline discovery-client/build.gradle:16]: should we define  `org.test-containers:spock` in `libs.versions.toml` ?

**Review by sdelamo — APPROVED** [review]

File: discovery-client/build.gradle (hunk 2 above)
Comment (sdelamo) [reply discovery-client/build.gradle]: done here: https://github.com/micronaut-projects/micronaut-discovery-client/pull/401/commits/cc93ca7dff369cdbbf0df21b7c6c1f05bfb58d82

**Pull request merged**